### DISCUSSION TITLE
Where does the 3KiB come from?

### DISCUSSION THREAD

**Discussion by webstrand** [preactjs/preact#4149] [General]
I've looked all over the place, and the closest I could find after brotli-ing `preact.min.js` at 4KiB. Is it just out of date, or is there a ultralight bundle that actually that small pre-compression?

**Comment by marvinhagemeister** [comment]
It's out of date. We should really update it to 4kB as it hovered around that size for a while now as we have kept adding compatibility fixes. There are some plans how to get that down to something <4kB again, but doing so would be a breaking change. That's the reason we've held off on doing so.

**Comment by effulgentsia** [comment]
Looks like currently it's around [4.7kB](https://github.com/preactjs/preact/pull/4618#issuecomment-2559153420), not counting `preact/hooks`, which adds another 1.5kB.

Also, https://preactjs.com/guide/v10/switching-to-preact says that preact/compat adds ~2kB, but the same link as above shows 4.1kB for `preact/compat` on top of the 1.5kB for `preact/hooks` which it depends on. In other words, the implication is that preact/compat is 5kB total (3+2), but really it's 10.3kB total.

Preact is a phenomenal library and worth its larger size compared to ancient versions, but yeah, it would be good to update those docs to reflect current numbers.

**Reply by rschristian** [reply]
Feel free to PR the compat size to our docs, happy to bump that up to ~4.2kb.

Hooks are a bit of an issue though, we don't include them in the main marketing size as they're completely optional but compat, being side-effectful, can't drop them at the moment.
